libs/SalesforceSDK/src/com/salesforce/androidsdk/push/PushService.kt

```diff
@@ -477,53 +482,55 @@ open class PushService {
                 .putString("ACTION", action.name)
                 .build()
 
-            when (pushNotificationsRegistrationType) {
-                ReRegistrationDisabled -> {
-                    /* Intentionally Blank */
-                }
-
-                ReRegistrationOnAppForeground -> OneTimeWorkRequest.Builder(
-                    workerClass = PushNotificationsRegistrationChangeWorker::class.java
-                ).setInputData(workData)
-                    .setConstraints(constraints)
-                    .build().also { workRequest ->
-                        workManager.enqueueUniqueWork(
-                            PUSH_NOTIFICATIONS_REGISTRATION_WORK_NAME,
-                            REPLACE,
-                            workRequest
-                        )
+            if (action == Register) {
```

Comment:
This is actually the main issue.  

The unified diff looks weird due to indentation but the change here is simply wrapping this `when (pushNotificationsRegistrationType)` in an `if (action == Register)` so we don't enqueue a **deregister** job here with the pushNotifications**Registration**Type and another (expected) deregister job below.